### DISCUSSION TITLE
Can Task Navigation be Added?

### DISCUSSION THREAD

**Discussion by aljawaid** [aljawaid/Glancer#26] [Q&A] [answered]
@creecros Can we add this?  A back (previous numbered task in project) / next (following numbered task in project)

https://github.com/kenlog/KangarooJump/issues/3

This would save the user from going back to the board etc to get into every task if he creates tasks in bulk as they would be numbered in order.  Even if not, the next button would go through each task to the project.

I'm guessing this would use `Pagination` but looking at the `TaskListController` I have absolutely no idea where to start.

**Comment by creecros** [comment]
yes, this should be relatively easy.

you need to use the TaskFinderModel, and the TaskViewController, with some knowledge of PHP. Thats pretty much it.

**Comment by creecros** [comment, marked as answer]
so as an example, I'll let you make it pretty, just add a hook:
`$this->template->hook->attach('template:task:show:top', 'glancer:layout/glancer_nav');`

and add the template file to hook:
```
<?php 
$alltasks = $this->model->taskFinderModel->getAllIds($project['id']); 
$key = array_search($task['id'], $alltasks); 
$last_key = count($alltasks)-1;
$next_key = ($key + 1 <= $last_key) ? $key + 1 : 0;
$prev_key = ($key - 1 >= 0) ? $key - 1 : $last_key;

$next = $alltasks[$next_key];
$prev = $alltasks[$prev_key];

?>
<?= $this->url->icon('arrow-right', t('Next Task'), 'TaskViewController', 'show', array('task_id' => $next)) ?>
<?= $this->url->icon('arrow-left', t('Prev Task'), 'TaskViewController', 'show', array('task_id' => $prev)) ?>
```

**Reply by aljawaid** [reply]
great, thanks, I shall work on this tonight

**Comment by aljawaid** [comment]
![image](https://user-images.githubusercontent.com/10233708/212473571-70884092-a5c1-429f-887f-459f1d5954f9.png)
